packages/audioplayers_platform_interface/lib/streams_interface.dart

```
@@ -20,6 +20,10 @@ mixin StreamsInterface {
     _positionStreamController.add(ForPlayer(playerId, value));
   }
 
+  void emitError(String playerId, String value) {
```

i'm not sure about this change, vs just logging the errors using the logger.
it concerns me that errors can go unoticed if people are not checking this stream.
also if we want to change probably need to change all instances of error logging.
is this necessary for this PR? can we break this off for more discussion?

---

Of course, I wanted to break it off anyways. But currently I have an error on Android, which only happens in the CI and I wanted to include more logs to the error on the dart side, which is only possible through an endpoint.
I noted for myself that it would be good to provide an endpoint to the user so he can react to such errors, or implement their own logging. The current behavior is not flexible enough in my opinion. Maybe it would be a good approach to have a standard output, that can be overwritten by the developer.
I also did not yet compare to other native packages, what they are doing.

These new features are still WIP. I'm very interested in your thoughts 😊 thank you.

---

Flutter VideoPlayer also provides internally a channel to handle platform errors. They are thrown, if the source is not yet initialized:
https://github.com/flutter/plugins/blob/c5220efae56f0f61cd2daebc6c7b5c8c088ef95f/packages/video_player/video_player/lib/video_player.dart#L420
So they could be catched on the dart side, so the theory. If source is initialized the error is simply ignored, I guess:
https://github.com/flutter/plugins/blob/c5220efae56f0f61cd2daebc6c7b5c8c088ef95f/packages/video_player/video_player/lib/video_player.dart#L420

As we don't have one channel for all AP event types, I would recommend to simply add an error channel. The native errors should be thrown anyways while calling `setSource`etc.

---

I splitted the error handling from this PR. We can move the discussion to #1352